string_permutation_problem/permute.py

```diff
@@ -0,0 +1,23 @@
+def permute_my_string(str):
+    output_string = []
+    """
+        This is a recursive solution to the string permutation problem.
+        The user is supposed to return all the permutations for a given string.
+    """
+    #Base Case
```

Comment:
The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpu5cefre7/string_permutation_problem/permute.py
+++ b/tmp/tmpu5cefre7/string_permutation_problem/permute.py
@@ -8,11 +8,11 @@
     if len(str) == 1:
         output_string = [str]
     else:
-        #For every letter in String
+        # For every letter in String
         for i, letter in enumerate(str):
-            #for every permutation resulting from Step 2 and 3 described above
+            # for every permutation resulting from Step 2 and 3 described above
             for perm in permute(str[:i] + str[i+1:]):
-                #Add it to output
+                # Add it to output
                 output_string += [letter + perm]
     return output_string
 
```